pkg/codegen/nodejs/gen.go

```diff
@@ -397,12 +397,14 @@ func printComment(w io.Writer, comment, deprecationMessage, indent string) {
 	fmt.Fprintf(w, "%s */\n", indent)
 }
 
-func (mod *modContext) genPlainType(w io.Writer, name, comment string, properties []*schema.Property, input, readonly bool, level int) {
+func (mod *modContext) genPlainType(w io.Writer, name, comment string,
```

Comment:
Would it be possible to factor out the `defaults` generation into a separate func? I'm squinting at this and I don't see how the `defaults` computation and code emission does not interact with the previous func much if at all.

Also.. `genPlainType` had too many bool parameters before the change, but now after the change with `plainType: bool` parameter it does strike me as very odd, if you're doing `genPlainType`, what does `plainType: false` mean now?

Reply:
I can work on that refactor. I think it would make sense. 

> Also.. genPlainType had too many bool parameters before the change, but now after the change with plainType: bool parameter it does strike me as very odd, if you're doing genPlainType, what does plainType: false mean now?

That's a valid point. We use `genPlainType` to generate argument types as well, so this was the "its really just a plain type" flag. I will see if I can think up a better name and will document what it does.